### ISSUE TITLE
[bitnami/postgres-ha] - failover issues when restarting worker node

### ISSUE THREAD

**Issue by giladneiger** [bitnami/charts#6554]
**Which chart**:
bitnami/postgres-ha:7.6.0

**Describe the bug**
TL;DR: pgpool crashing & restarting after restarting the (postgres) primary worker node

We have tried to simulate a worker node failure (reboot) and expected to still have postgres database access as we are running HA.
Instead of that, the pgpool kept restarting itself and we have seen multiple (2) primary replicas.

![image](https://user-images.githubusercontent.com/40278137/120661694-2da4a900-c491-11eb-9d08-407976887215.png)

The are the logs of the pgpool pod while it crashed:

![image](https://user-images.githubusercontent.com/40278137/120665042-1b783a00-c494-11eb-9f48-3d5c770749af.png)

**To Reproduce**

1. We used the following image tags: 
         - bitnami/pgpool:4.2.3-debian-10-r2
         - bitnami/postgresql-repmgr:12.6.0-debian-10-r85
2. Install helm chart with postgres replicas count set to 3
3. SSH to the worker node that runs the primary replica and reboot it
4. `kubectl exec` to a postgres replica and run the following command:
    `/opt/bitnami/scripts/postgresql-repmgr/entrypoint.sh repmgr -f /opt/bitnami/repmgr/conf/repmgr.conf cluster show`
    You will probably see multiple primaries. If not - try running the command again. It's inconsistent. 
5. While the node is down (can take a few minutes), run `kubectl logs` on the pgpool pod. You'll see that it is restarting its child processes:

![image](https://user-images.githubusercontent.com/40278137/120663128-6f821f00-c492-11eb-999f-a3bd872eddf3.png)

**Expected behavior**
We expected to have postgres database access through the pgpool service as we are running HA. We didn't get it because pgpool kept restarting. We also expected to not have 2 primaries.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.1", GitCommit:"fe51cd1e31e6a202cba7dead9552a6d418ded79a", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by rafariossaa** [comment]
Hi,
  The pgpool messages are normal because pgpool kills the process when not  longer needed.
  About the 2 primary posgresql , could you provide the logs form the node-0 and other secundary node ? I would like to check if this could be something related with the timing. Maybe the node-0 restarts very fast meanwhile the other nodes are in the promoting steps.
  I got different results, I deployed in GCE a k8s cluster and then deployed a postgresql-ha cluster with the fololwing command
```
helm install mypg -f values.yaml --set postgresql.password=test --set postgresql.replicaCount=3 .
````
(I am using the default `values.yaml`).
  Once the cluster is started I checked the status and was ok, then I killed the pod with: 
```
$ kubectl delete pod mypg-postgresql-ha-postgresql-0 
```
  Meanwhile the pod was rebooting I got:
```
 ID   | Name                            | Role    | Status    | Upstream                          | Location | Priority | Timeline | Connection string                                                                                                                                                              
------+---------------------------------+---------+-----------+-----------------------------------+----------+----------+----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1000 | mypg-postgresql-ha-postgresql-0 | primary | ? running | ?                                 | default  | 100      |          | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-0.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5
 1001 | mypg-postgresql-ha-postgresql-1 | standby |   running | ! mypg-postgresql-ha-postgresql-0 | default  | 100      | 1        | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-1.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5
 1002 | mypg-postgresql-ha-postgresql-2 | standby |   running | ! mypg-postgresql-ha-postgresql-0 | default  | 100      | 1        | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-2.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5

WARNING: following issues were detected
  - unable to connect to node "mypg-postgresql-ha-postgresql-0" (ID: 1000)
  - node "mypg-postgresql-ha-postgresql-1" (ID: 1001) is not attached to its upstream node "mypg-postgresql-ha-postgresql-0" (ID: 1000)
  - node "mypg-postgresql-ha-postgresql-2" (ID: 1002) is not attached to its upstream node "mypg-postgresql-ha-postgresql-0" (ID: 1000)
```
  which is normal.
  After the reboot:
```
 ID   | Name                            | Role    | Status               | Upstream                        | Location | Priority | Timeline | Connection string                                                                                                                                                              
------+---------------------------------+---------+----------------------+---------------------------------+----------+----------+----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1000 | mypg-postgresql-ha-postgresql-0 | primary | * running            |                                 | default  | 100      | 1        | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-0.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5
 1001 | mypg-postgresql-ha-postgresql-1 | standby | ! running as primary |                                 | default  | 100      | 2        | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-1.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5
 1002 | mypg-postgresql-ha-postgresql-2 | standby |   running            | mypg-postgresql-ha-postgresql-0 | default  | 100      | 1        | user=repmgr password=yMQDBtL8LN host=mypg-postgresql-ha-postgresql-2.mypg-postgresql-ha-postgresql-headless.default.svc.cluster.local dbname=repmgr port=5432 connect_timeout=5

WARNING: following issues were detected
  - node "mypg-postgresql-ha-postgresql-1" (ID: 1001) is registered as standby but running as primary
```
  That shouldn't be happening, but it is not the same you got.
